Paper: Global optimization software library for research and education

nadiakap opened the pull request:
*Editor*: <!--editor-->@deniederhut<!--end-editor-->

*Reviewers*: <!--reviewers-list-->@himaghna<!--end-reviewers-list-->

nadiakap committed:
Add files via upload

deniederhut committed:
BUG: fixed a few typos so paper builds

nadiakap committed:
Merge pull request #1 from deniederhut/bug/nadiakap-paper

BUG: fixed a few typos so paper builds

scoobies commented:
Hi!, I'm @scoobies, a friendly bot.

Type ```@scoobies help``` to discover how I can help you.

deniederhut commented:
@scoobies mark paper

deniederhut commented:
@scoobies mark needs review

deniederhut commented:
@scoobies build status

scoobies commented:
{"nadiakap-2022"=>{"data"=>{"build_output"=>"[*] Updating proceedings build tools in {self.master_repo_path}...\nAlready up to date.\n[*] Check out paper repository...\nCloning into '/tmp/tmpy1qgafn2'...\nMoving proceedings build tools to temp directory...\n[*] Build the paper...\nBuilding: nadiakap\n*** Warning: /tmp/tmpy1qgafn2/publisher/../output/nadiakap/paper_stats.json does not exist.\n", "build_pdf_path"=>"/procbuild/procbuild/../cache/nadiakap-2022.pdf", "build_status"=>"success", "build_timestamp"=>"22/05 17:43"}, "status"=>"success"}}

deniederhut commented:
@scoobies assign @deniederhut as editor

scoobies commented:
Assigned! @deniederhut is now the editor

deniederhut commented:
Hi @nadiakap ! I'm looking at the copy of your paper on the build server and it just a little bit over one page. Was there more you were planning on writing?

mepa commented:
@scoobies add @himaghna to reviewers

scoobies commented:
@himaghna added to the reviewers list!

deniederhut commented:
Hi! We've moved the deadline for first reviews back one week, to June 16th, to give us a little more time to find reviewers for all the papers this year. You can see the updated full timeline at https://github.com/scipy-conference/scipy_proceedings#timeline-for-2022.

nadiakap commented:
looks like formulas in my paper are not compiling. they are in .tex format. how do i convert them to .rst? or may be there is automatic conversion tool? i cannot find .rst documentation on typin math symbols such as integrals, sums, ..
would you please let me know how i should proceed?

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Merge branch 'scipy-conference:2022' into 2022

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

added three more sections

nadiakap committed:
Update mypaper.rst

nadiakap commented:
corrected math in my paper and want to push to remote repository but git push fails (asking for authentication code and fails when I enter it) 
committed my changes to nadiakap/scipy_proceedings
not sure how to git oush to scipy_conference/scipy_proceedings

nadiakap committed:
Update mypaper.rst

deniederhut committed:
BUG: fix whitespace around directives

deniederhut commented:
Hey @nadiakap ! When you push to your fork (nadiakap/scipy_proceedings) it automatically shows up here. The thing keeping your paper from building was some extra whitespace characters. I fixed those (and a couple other formatting things) in this PR here -- https://github.com/nadiakap/scipy_proceedings/pull/3 -- and confirmed that it builds locally. If you click the merge button in your fork, those changes will show up here, and your paper will build again.

nadiakap committed:
Merge pull request #3 from deniederhut/bug/whitespace

BUG: fix whitespace around directives

nadiakap commented:
thank you @deniederhut! I see paper builds

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

fixed references and some math

nadiakap committed:
Update mypaper.rst

nadiakap commented:
fixed some references and formulas. cannot figure out how to merge these changes.

nadiakap committed:
Update mypaper.rst

fixed reference  errors

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

some references fixes

nadiakap committed:
Update mypaper.rst

deniederhut commented:
> cannot figure out how to merge these changes

I see some commits from yesterday -- did you figure out the merging issue?

deniederhut commented:
Oh, and @himaghna -- how is the review coming along?

himaghna commented:
@deniederhut Thanks for assigning me to this review! I should be done with my review in a day. Quick introduction; I am Himaghna Bhattacharjee, a fifth year PhD student in computational chemical engineering at the Vlachos lab in University of Delaware. I am currently interning as an ML PhD Intern at Meta.

nadiakap commented:
@deniederhut - the merge worked ! thank you!

nadiakap committed:
Update mypaper.rst

nadiakap commented:
@himaghna - thank you for reviewing!

I did not know in advance how lonf the paper is in .rst format, so I made it bas short as possible. i can see that it is only three pages, i am planning to add more details ( code pieces/flowcharts) under 'practical considerations' section if this is till ok

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

added algorithm building blocks description in 'practical considerations' section

deniederhut commented:
> I should be done with my review in a day.

Awesome

> I am currently interning as an ML PhD Intern at Meta.

Congrats!

> the merge worked

Also awesome

nadiakap committed:
Update mypaper.rst

removed something that I recently  added

deniederhut commented:
Hey @himaghna ! Friendly reminder that initial reviews are due tomorrow

nadiakap commented:
Hi @himaghna, nice meeting you! look forward to your feedback

himaghna commented:
Hi @nadiakap , before I finish my review and FYI, I notice that the submission pdf is still 3 pages. Did you get a chance to add any of the changes you had wanted?

nadiakap commented:
Hi @himaghna, decided to have 3 pages for now,
I wanted to add some code examples but decided to leave it as it is is.
 I have user documentation and working on 'getting started with minpy' jupiter notebook.  these docuemnts will have code examples

himaghna commented:
> Hi @himaghna, decided to have 3 pages for now, I wanted to add some code examples but decided to leave it as it is is. I have user documentation and working on 'getting started with minpy' jupiter notebook. these docuemnts will have code examples

Sounds good!

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Create mypaper.rst

nadiakap committed:
Create mypaper.rst

nadiakap commented:
there are  typos, fixing them now!

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

himaghna commented:
Hi @nadiakap  thanks for contributing the paper! My high level thoughts are detailed below:

Overall, the paper and the software has the potential to be super exciting to the community, especially if the educational aspects are considered. The method outlined which avoids the need for derivatives of the black-box function is very interesting. We believe the following suggestions and comments can improve the readability of the paper and provide a better context to the readers:

1. A discussion on and comparison to other existing tools and software which provide similar functionality in the introductory sections would be very helpful to place this work in context.
2. Since this is a primarily software paper, a more elaborate discussion on the organization, features and limitations of the software would be helpful.
3. Clearing up grammatical and typographical error would greatly enhance the accessibility of the paper.

himaghna commented on papers/nadiakap/mypaper.rst:
```diff
@@ -0,0 +1,281 @@
+Optimization lies at the heart of machine learning and data science.
+One of the most relevant problems in machine learning is automatic selection of the algorithm depending on
+the objective. This is necessary in many applications such as robotics, simulating biological or chemical
+processes, trading strategies optimization, to name a few.
```
Reference to this claim lacking

himaghna commented on papers/nadiakap/mypaper.rst:
```diff
@@ -0,0 +1,281 @@
+
+One typical situation is when different algorithm behavior is required when being far from optimal point, and when being close to optimal point.
+
+As an example, consider variant of simulated annealing proposed in [FGSB]_ where different values of parameters ( Boltzman distribution parameters, step size, etc.) are used depending of the distance to optimal point. In this paper the basic SA algorithm is used as a starting point. We can offer more basic module as a starting point ( and by specifying distribution as 'exponential' get the varian ...
```
Define the acronym "SA"

himaghna commented on papers/nadiakap/mypaper.rst:
```diff
@@ -0,0 +1,281 @@
+
+The approach works for non-smooth or algorithmically defined functions.  For detailed description of the approach see [KAP]_, [KP]_.
+In this approach the original optimization problem is replaced with a randomized problem, allowing the use of Monte-Carlo methods for calculating integrals.
+This is especially important if objective function is given by its values (no analytical formula) and derivatives
```
"the" objective function..

himaghna commented on papers/nadiakap/mypaper.rst:
```diff
@@ -0,0 +1,281 @@
+
+To study the gradient nature of the solution algorithms for problem :ref:`rfx`, a variation of objective functional  F(X)  will be considered.
+
+The suggested approach makes it possible to obtain optimization methods in systematic way, similar to methodology adopted in smooth optimization. 
```
nit: "the methodology..

himaghna commented on papers/nadiakap/mypaper.rst:
```diff
@@ -0,0 +1,281 @@
+Stochastic extention of Nelder and Mead algorithm
+-------------------------------------------------
+
+The following algorithm was obtained using general steps describded in the previous section  and assuming uniform distribution in randomization process.
```
nit: type in the word "described"

himaghna commented on papers/nadiakap/mypaper.rst:
```diff
@@ -0,0 +1,281 @@
+The library is intended primarily for educational
+purposes and its focus is on transparency of the methods rather than on efficiency of implementation.
+
+Minpy can be used by researches to design optimization methods with desired properties by varying parameters of the general algorithm. Such design step is very often required in newly developing industry projects, where objective function has unknown behavior, Algorithm designing stage is necessary for two reasons: It helps to understand objective function behavior, and it is needed for function  ...
```
nit: unknown behavior. An algorithm design stage..

himaghna commented on papers/nadiakap/mypaper.rst:
```diff
@@ -0,0 +1,281 @@
+
+If approximate :math:`\epsilon _{k-1}` cannot be obtained within the specified number of trails, then set :math:`m_k=m_{k-1}`
+
+Use the sample standard deviation as the termination criterion: :math:`D_k=(\frac{1}{K-1} \sum_{i=1}^K (f(u_k^i)-c_k)^2)^{1/2}`
```
It might be useful to think of key takeaways for the user: How do I use this software and leverage its capability for my use-case?

nadiakap replied:
Hi @himaghna. here is key takeaways for the user: may be it should take a whole new section in the paper?
Many industry professionals and researchers utilize mathematical optimization packages to search for better solutions of their problems. Examples of such problem include minimization of free energy in physical system [2] or robot gait optimization from robotics [1], or designing materials for 3d printing [4], or optimization of chemical reaction [3]. These problems may involve "black box optimization", where the structure of the objective function is unknown and is revealed through a small sequence of expensive trials. Software implementations for these methods become more user friendly. As a rule, however, certain modeling skills are needed to formulate real world problem in a way suitable for applying software package. Moreover, selecting optimization method appropriate for the model is a challenging task. 
Our educational software helps users of such optimization packages and may be considered as a companion to them. The focus of our software is on transparency of the methods rather than on efficiency. A principal benefit of our software is the unified approach for constructing algorithms whereby any other algorithm is obtained from the generalized algorithm by changing certain parameters. Well known heuristic algorithms such as Nelder and Mead (NM) algorithm may be obtained using this generalized approach, as well as new algorithms. Although some derivative-free optimization packages (matlab global optimization toolbox,  Tensorflow Probability optimizers, Excel Evolutionary Solver, scikit-learn Stochastic Gradient Descent class) put a lot of effort in transparency and educational value, they don't have the same level of flexibility and generality as our system. 

[1] A Hybrid EDA/Nelder-Mead for Concurrent Robot Optimization
https://www.researchgate.net/publication/331902083_A_Hybrid_EDANelder-Mead_for_Concurrent_Robot_Optimization

[2]Random reselection particle swarm optimization for optimal design of solar photovoltaic modules
https://www.sciencedirect.com/science/article/abs/pii/S0360544221021137

[3]Self-optimising processes and real-time-optimisation of organic syntheses in a microreactor system using Nelder–Mead and design of experiments
https://pubs.rsc.org/en/content/articlelanding/2020/re/d0re00081g

[4]Characterisation and Comparison of Material Parameters of 3D-Printable Absorbing Materials
https://europepmc.org/article/med/35208040#free-full-text

himaghna reviewed (changes requested):
Revisions requested based on the inline and high level comments

deniederhut commented:
Thanks Himaghna!

deniederhut commented:
@scoobies mark pending comment

nadiakap commented:
@Himaghna, thank you for your comments.
I will make the requested changes and submit again.

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

nadiakap commented:
made some corrections, and the build failed.  @deniederhut,  how do I look at the log to see what exactly is wrong?

nadiakap committed:
Update mypaper.rst

deniederhut commented:
@scoobies build status

scoobies commented:
{"nadiakap-2022"=>{"data"=>{"build_output"=>"[*] Updating proceedings build tools in {self.master_repo_path}...\nAlready up to date.\n[*] Check out paper repository...\nCloning into '/tmp/tmp0_96lrop'...\nMoving proceedings build tools to temp directory...\n[*] Build the paper...\nBuilding: nadiakap\n*** Warning: /tmp/tmp0_96lrop/publisher/../output/nadiakap/paper_stats.json does not exist.\nPDFLaTeX error output:\n================================================================================\nThis is pdfTeX, Version 3.14159265-2.6-1.40.18 (TeX Live 2017/Debian) (preloaded format=pdflatex)\n restricted \\write18 enabled.\nentering extended mode\n(./paper.tex\nLaTeX2e <2017-04-15>\nBabel <3.18> and hyphenation patterns for 3 language(s) loaded.\n(./IEEEtran.cls\nDocument Class: IEEEtran 2015/08/26 V1.8b by Michael Shell\n-- See the \"IEEEtran_HOWTO\" manual for usage information.\n-- http://www.michaelshell.org/tex/ieeetran/\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/ot1ppl.fd)\n-- Using IEEE Computer Society mode.\n-- Using 8.5in x 11in (letter) paper.\n-- Using PDF output.\n-- This is a 10 point document.\n) (/usr/share/texlive/texmf-dist/tex/latex/cmap/cmap.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/base/ifthen.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/base/fontenc.sty\n(/usr/share/texlive/texmf-dist/tex/latex/base/t1enc.def)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/t1ppl.fd)<<t1.cmap>>)\n(/usr/share/texlive/texmf-dist/tex/latex/base/inputenc.sty\n(/usr/share/texlive/texmf-dist/tex/latex/base/utf8.def\n(/usr/share/texlive/texmf-dist/tex/latex/base/t1enc.dfu)\n(/usr/share/texlive/texmf-dist/tex/latex/base/ot1enc.dfu)\n(/usr/share/texlive/texmf-dist/tex/latex/base/omsenc.dfu)))\n(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amsmath.sty\nFor additional information on amsmath, use the `?' option.\n(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amstext.sty\n(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amsgen.sty))\n(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amsbsy.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amsopn.sty))\n(/usr/share/texlive/texmf-dist/tex/latex/base/textcomp.sty\n(/usr/share/texlive/texmf-dist/tex/latex/base/ts1enc.def\n(/usr/share/texlive/texmf-dist/tex/latex/base/ts1enc.dfu))) (./scipy.sty\n(./status.sty (./draftwatermark.sty (./everypage.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphicx.sty\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/keyval.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphics.sty\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/trig.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/graphics-cfg/graphics.cfg)\n(/usr/share/texlive/texmf-dist/tex/latex/graphics-def/pdftex.def)))\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/color.sty\n(/usr/share/texlive/texmf-dist/tex/latex/graphics-cfg/color.cfg))))\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/mathptmx.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/helvet.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/courier.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/amsfonts/amsfonts.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/amsfonts/amssymb.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/tools/bm.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/hyperref/hyperref.sty\n(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/hobsub-hyperref.sty\n(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/hobsub-generic.sty))\n(/usr/share/texlive/texmf-dist/tex/generic/ifxetex/ifxetex.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/auxhook.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/kvoptions.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/hyperref/pd1enc.def)\n(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/hyperref.cfg)\n(/usr/share/texlive/texmf-dist/tex/latex/url/url.sty))\n(/usr/share/texlive/texmf-dist/tex/latex/hyperref/hpdftex.def\n(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/rerunfilecheck.sty))\n\nPackage hyperref Warning: Option `pdftex' has already been used,\n(hyperref)                setting the option has no effect on input line 43.\n\n(/usr/share/texlive/texmf-dist/tex/latex/tools/longtable.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/multirow/multirow.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/booktabs/booktabs.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/upquote/upquote.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/fancyvrb/fancyvrb.sty\nStyle option: `fancyvrb' v2.7a, with DG/SPQR fixes, and firstline=lastline fix \n<2008/02/07> (tvz)))\nNo file paper.aux.\n(/usr/share/texlive/texmf-dist/tex/latex/base/ts1cmr.fd)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/t1ptm.fd)\n-- Lines per column: 61 (exact).\n(/usr/share/texlive/texmf-dist/tex/context/base/mkii/supp-pdf.mkii\n[Loading MPS to PDF converter (version 2006.09.02).]\n) (/usr/share/texlive/texmf-dist/tex/latex/oberdiek/epstopdf-base.sty\n(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/grfext.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg))\n(/usr/share/texlive/texmf-dist/tex/latex/hyperref/nameref.sty\n(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/gettitlestring.sty))\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/t1phv.fd)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/ot1ztmcm.fd)<<ot1.cmap>>\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/omlztmcm.fd)<<oml.cmap>>\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/omsztmcm.fd)<<oms.cmap>>\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/omxztmcm.fd)<<omx.cmap>>\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/ot1ptm.fd)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/upzd.fd)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/ts1ptm.fd) (./page_numbers.tex)\n\nLaTeX Warning: Citation `KAP' on page 1 undefined on input line 159.\n\n\nLaTeX Warning: Citation `KHNT' on page 1 undefined on input line 174.\n\n\nLaTeX Warning: Citation `FGSB' on page 1 undefined on input line 188.\n\n\nLaTeX Warning: Citation `GLUQ' on page 1 undefined on input line 188.\n\n\nLaTeX Warning: Citation `KAP' on page 1 undefined on input line 193.\n\n\nLaTeX Warning: Citation `KPP' on page 1 undefined on input line 193.\n\n\nLaTeX Warning: Citation `BIS' on page 1 undefined on input line 195.\n\n\nLaTeX Warning: Citation `NBAG' on page 1 undefined on input line 196.\n\n\nLaTeX Warning: Citation `ZALO' on page 1 undefined on input line 196.\n\n\nLaTeX Warning: Citation `ATFB' on page 1 undefined on input line 197.\n\n\nLaTeX Warning: Citation `KAP' on page 1 undefined on input line 204.\n\n\nLaTeX Warning: Citation `KP' on page 1 undefined on input line 204.\n\n[1{/var/lib/texmf/fonts/map/pdftex/updmap/pdftex.map}]\n\nLaTeX Warning: Reference `rfx' on page 2 undefined on input line 225.\n\n\nLaTeX Warning: Reference `fx' on page 2 undefined on input line 225.\n\n\nLaTeX Warning: Reference `rfx' on page 2 undefined on input line 225.\n\n! Missing \\endcsname inserted.\n<to be read again> \n                   \\protect \nl.226 Note that :ref:`rfx}\n                           is the stochastic optimization problem of the fun...\n!  ==> Fatal error occurred, no output PDF file produced!\nTranscript written on paper.log.\n\n================================================================================\n[X] Paper build failed.\nA build error occurred during the self.retrieve_pdf step.", "build_pdf_path"=>"", "build_status"=>"Build failed, no pdf can be found", "build_timestamp"=>"24/06 00:31"}, "status"=>"fail"}}

deniederhut commented on papers/nadiakap/mypaper.rst:
```diff
@@ -0,0 +1,292 @@
+
+where  X is a random vector from :math:`R^n`, {X} is a set of such random vectors, and :math:`E[\cdot]` is the expectation operator. 
+
+Problem  :ref:`rfx` is equivalent to problem :ref:`fx` in the sense that any realization of the random vector :math:`X^*`, where :math:`X^*` is a solution to :ref:`rfx`, that has a nonzero probability, will be a solution to problem :ref:`fx`(see [KAP]_ for proof). 
```
```suggestion
Problem  :ref:`rfx` is equivalent to problem :ref:`fx` in the sense that any realization of the random vector :math:`X^*`, where :math:`X^*` is a solution to :ref:`rfx`, that has a nonzero probability, will be a solution to problem :ref:`fx` (see [KAP]_ for proof). 

```

deniederhut commented:
@nadiakap there was a missing space after one of the backticks

nadiakap committed:
Update papers/nadiakap/mypaper.rst

Co-authored-by: Dillon Niederhut <deniederhut@users.noreply.github.com>

nadiakap commented:
@deniederhut], thank you for corrections!

deniederhut commented:
🙇‍♂️

deniederhut commented:
@nadiakap are you still working on those revisions? Or is the paper ready for re-review?

nadiakap committed:
Update mypaper.rst

nadiakap commented:
@deniederhut, I added couple of lines today and it is erroring out again. please  point me to the log if you can
not planning to do any more changes
@Himaghna,  - not sure I addressed your first comment correctly, left a question there. Thanks

nadiakap committed:
Update mypaper.rst

nadiakap commented:
@deniederhut,, errors are fixed,

nadiakap commented:
@himaghna,  - realized finally how to address your first comment better, 
@deniederhut, - may I put one more update, to address @himaghna first comment?

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

small fix - misspelled word

deniederhut commented:
Glad you got it figured out! @himaghna, will you be able to give this paper a second look this week?

himaghna commented:
Hi! Thanks for the revisions @nadiakap! I will have another set of reviews
ready by the end of the week.

On Sun, Jun 26, 2022 at 8:16 PM Dillon Niederhut ***@***.***>
wrote:

> Glad you got it figured out! @himaghna <https://github.com/himaghna>,
> will you be able to give this paper a second look this week?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/scipy-conference/scipy_proceedings/pull/684#issuecomment-1166788730>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIMLVY5HOIG3GXQA6RC7KBLVREMKVANCNFSM5WAVCSUA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Himaghna Bhattacharjee
George W. Laird Fellow,
Ph.D. Candidate, Vlachos Research Group
Chemical and Biomolecular Engineering
University of Delaware
Phone: 917 488 0877

*“**You have to want to falsify the model. If you love somebody, set them
free.”* *- Gelman*

nadiakap commented:
@himaghna, the jupyter notebook to accompany the paper  is almost ready. may be it should be in appendix in the paper? is this ok to add the appendix now? or may be it is better to share the notebook separately?

@deniederhut what is the best way to share the notebook?

thank you!

deniederhut commented:
Hm maybe you could link to a binder instance of the notebook in the paper?

nadiakap commented:
@deniederhut, thank you, will look at this

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Create mypaper.rst

intead of adding link to Jupitr notebook added more things to the last section of the paper

nadiakap committed:
Update mypaper.rst

nadiakap commented:
added more discussion to the last section instead of adiing a link to jupiter notebook
but it fails now.

nadiakap committed:
Update mypaper.rst

nadiakap commented:
the error is fixed,

nadiakap committed:
Update mypaper.rst

deniederhut commented:
@himaghna how is the review coming along?

deniederhut commented on papers/nadiakap/mypaper.rst:
Are we missing a volume and page number for this?

nadiakap replied:
this is how it is cited  in one Automation and Remote Control paper:
Kaplinskii, A.I. and Propoi, A.I., First-Order Nonlocal Optimization Methods based on Potential Theory, Avtom.Telemekh., 1994, no. 7, pp. 94–103.

nadiakap replied:
this is where it is cited
https://link.springer.com/article/10.1023/A%3A1020951230751

deniederhut replied:
Thanks! We want complete citations in the references section of the paper -- not the GitHub comments. Would you mind updating them?

deniederhut commented on papers/nadiakap/mypaper.rst:
Volume and pages for this one too?

nadiakap replied:
Hi @deniederhut, for [KPP] - this is what I found:
Kaplinskij, A.I. & Pesin, A.M. & Propoj, A.I.. (1994). Analysis of search methods of optimization based on potential theory. III: Convergence of methods. Automation and Remote Control. 1. 
page numbers are not provided.
this reference is taken from the following:
https://www.researchgate.net/scientific-contributions/AI-Kaplinskij-2052913184

deniederhut commented on papers/nadiakap/mypaper.rst:
Volume, page number, and DOI for this one?

nadiakap replied:
@deniederhut, 
here is the info:
Volume 15, (2019) Article 4 pp. 1-32
[DOI: 10.4086/toc.2019.v015a004](http://dx.doi.org/10.4086/toc.2019.v015a004)

deniederhut commented on papers/nadiakap/mypaper.rst:
Proceedings, volume, and page number for this one?

nadiakap replied:
@deniederhut, here is a link to the paper. I cited as suggested in the link https://arxiv.org/abs/1902.00947
would it still work?

deniederhut commented:
@nadiakap  would you mind taking a second pass over the references section? Some of these look incomplete

himaghna commented:
Hi @nadiakap  These changes look great. Thanks for incorporating my suggestions. The discussion section is much clearer and gives a better context of the software and its application. One final change I would suggest is to provide a small section on how this software may be accessed (GitHub link / installation instructions etc.). Finally, please address @deniederhut 's comment.

**My final recommendation is that pending these changes, this article be accepted for SciPy '22!**

nadiakap commented:
@himaghna, thank you for the review. I will add the link to public github repository for the library and the docs. I guess this is where getting started notebook should be

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

deniederhut commented:
Hey @nadiakap ! I see that you responded to a couple of my comments, but a lot of these reference are missing information or might otherwise be incorrect. Would you mind going through them all to make sure they are complete?

nadiakap commented:
@deniederhut,  verifying references again - need some more time.

nadiakap committed:
Update mypaper.rst

nadiakap commented:
@deniederhut, I can see volume but cannot find pages info  for this reference:

Kaplinskij, A.I. & Pesin, A.M. & Propoj, A.I.. (1994). Analysis of search methods of optimization based on potential theory. III: Convergence of methods. Automation and Remote Control. Volume 1

should this reference be removed from paper?

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

nadiakap commented:
@deniederhut, all references are corrected now - they are taken exactly as they are suggested to be cited. 
this is why they all have different format. is this ok or should they be of the same format?

nadiakap committed:
Update mypaper.rst

nadiakap committed:
Update mypaper.rst

nadiakap commented:
@deniederhut, made final corrections to first three references. it is taken from here:
https://archive.org/details/sim_automation-and-remote-control_1994_55_contents/page/n7/mode/2up

deniederhut commented on papers/nadiakap/mypaper.rst:
@nadiakap when I Google "Multiphase Simulated Annealing Based on Boltzmann and Bose-Einstein Distribution Applied to Protein Folding Problem", the first link is to https://www.hindawi.com/journals/abi/2016/7357123/, which displays a full citation with the volume number and DOI, both of which are missing from your references section.

deniederhut reviewed (changes requested):
Please check all of your references for completeness -- not just the ones where I have left comments.

nadiakap committed:
Update mypaper.rst

nadiakap commented:
@deniederhut, thank you ! it is corrected

deniederhut commented:
@nadiakap I still see incomplete references. Please check all of your references -- not just the ones where I have left comments.

nadiakap commented:
@deniederhut, I thought I checked everything, my apologies!! will try again

nadiakap committed:
Update mypaper.rst

nadiakap commented:
@deniederhut, added doi where it exists. for earlier papers (1994) and for the book published in University Publishing House doi did not exist.

deniederhut commented:
@nadiakap thanks I'll take a look. It looks like this paper is no longer building on the server -- I see two errors in the logs

1. Explicit markup ends without a blank line; unexpected unindent
2. Unknown target name: \"khnt\"

nadiakap commented:
@deniederhut, my apologies for taking so much time to complete things, but I need your help again.
would you please send me link for viewing logs of papers to see the compilation errors. 
I do not know how to view the log and without it cannot see how to fix the error.
it seems that the reference before [HLKT] (line 303) has some issues but I cannot understand what exactly is wrong

nadiakap committed:
Update mypaper.rst

nadiakap commented:
@deniederhut,  sorry, the problem is in line 330 not 303.
I tried to delete the extra space there but cannot see what else may be wrong.

nadiakap commented:
@deniederhut, I somehow lost the link to see compiled papers. would you please send it again. Thank you so much!

deniederhut commented:
@nadiakap no worries -- the build status for your paper is at https://procbuild.scipy.org/status/nadiakap-2022

deniederhut commented on papers/nadiakap/mypaper.rst:
@nadiakap looks like an extra newline sneaked in here?

nadiakap committed:
Update mypaper.rst

nadiakap commented:
@deniederhut, thank you, it builds now

deniederhut commented:
@nadiakap awesome! I think the last thing left to do here is add consistent DOI formatting so they follow the guidelines here https://www.crossref.org/display-guidelines/

nadiakap committed:
Update mypaper.rst

fixes doi format where doi is available

nadiakap commented:
@deniederhut - thank you to pointing to doi standard page! 
do references are fixed, and paper builds!
three older papers and one book do not have doi

deniederhut reviewed (approved):
Alright let's boogie
